src/lib/modules/api/models/historic-percentile-param.enum.ts

```diff
@@ -0,0 +1,13 @@
+
+export enum HistoricPercentileParam {
+  One = 1,
+  Five = 5,
+  NinetyFive = 95,
+  NinetyNine = 99
+}
+
+const keys = Object.keys(HistoricPercentileParam).filter(k => {
+  return typeof HistoricPercentileParam[k as any] === 'number';
```

Comment:
This stuck out to me as very robust haha. Curiosity, why did you write it like this?

Reply:
Can't claim original authorship over that strategy, it [came from here](https://github.com/Microsoft/TypeScript/issues/17198#issuecomment-315400819) as a suggestion by one of the TypeScript devs (I think).